feat: Add GitHub and Google sign-in buttons

sidarth-23 opened the pull request:
<!--
  Thanks for submitting a pull request!
  We appreciate you spending the time to work on these changes. Please provide enough information so that others can review your pull request. 
-->

## Summary
- Add GitHub sign-in button to allow users to sign in with GitHub.
- Add Google sign-in button to allow users to sign in with Google.
- These buttons will authenticate users and set cookies for access and refresh tokens.
- Added new types for Google Sign-In button and Google Sign-In function.
- Updated Google Sign-In button component to use the new interfaces.
- Updated global window declaration for Google Sign-In functionality.
<!--
 Explain the **motivation** of this Pull Request
-->

- [x] Tested (Must)
- [ ] Test Case added
- [x] Build Successful (Must)
- [ ] Sufficient Code comments added (Must)
- [x] Attached Screenshots / Videos <!-- if the PR contains UI changed, fixes, improvements -->
- [x] All Relevant Documents added

### Screenshot
![image](https://github.com/kalvilabs/kalvi/assets/58748190/82ec545e-d56b-434d-a909-e841683a62f1)


## Depends on
#97 
<!--- Does this PR depend on another PR that should be merged first or at the same time -->

elizabeththomas92 commented on apps/admin/components/google-signin.tsx:
```diff
@@ -0,0 +1,110 @@
+    if (!googleSignInButton.current || gsiScriptLoaded) return;
+    console.log(process.env.GOOGLE_CLIENT_ID!);
+
+    (window as any)?.google?.accounts.id.initialize({
```
I see this being used at some places repeatedly ,maybe destructure and use directly? 

`const googleId = window?.google?.accounts?.id;`


```
googleId.initialize({
      client_id: clientId,
      callback: handleGoogleSignIn,
    });
```

yaju09 commented on apps/admin/components/github-signin.tsx:
```diff
@@ -0,0 +1,68 @@
+      } else throw Error("Cant find credentials");
+    } catch (err: any) {
+      // TODO: Handle error
+      console.error(err);
```
```suggestion
      console.error(err);
```
please remove console

yaju09 commented on apps/admin/components/github-signin.tsx:
```diff
@@ -0,0 +1,68 @@
+  };
+  const searchParams = useSearchParams();
+  // states
+  const [loginCallBackURL, setLoginCallBackURL] = useState(undefined);
```
```suggestion
  const [loginCallBackURL, setLoginCallBackURL] = useState();
```
is undefined is to be given explicitly?

yaju09 commented on apps/admin/components/github-signin.tsx:
```diff
@@ -0,0 +1,68 @@
+      typeof window !== "undefined" && window.location.origin
+        ? window.location.origin
+        : "";
+    setLoginCallBackURL(`${origin}/` as any);
```
```suggestion
    setLoginCallBackURL(`${origin}/` as any);
``
is it possible to handle any?`

yaju09 commented on apps/admin/components/google-signin.tsx:
```diff
@@ -0,0 +1,110 @@
+export const GoogleSignInButton: FC<Props> = () => {
+  const router = useRouter();
+  const clientId = process.env.NEXT_PUBLIC_GOOGLE_CLIENT_ID;
+  const handleGoogleSignIn = async ({ clientId, credential }: any) => {
```
```suggestion
  const handleGoogleSignIn = async ({ clientId, credential }: any) => {
```
is it possible to address any?

sidarth-23 replied:
I think that is bit difficult for now

yaju09 commented on apps/admin/components/google-signin.tsx:
```diff
@@ -0,0 +1,110 @@
+      router.push("/admin/dashboard");
+    } catch (err) {
+      const error = err as AxiosError<TGoogleSignInErrorResponse>;
+      console.error(error);
```
```suggestion
      console.error(error);
```
is the above console required?

yaju09 commented on apps/admin/components/google-signin.tsx:
```diff
@@ -0,0 +1,110 @@
+        } as GsiButtonConfiguration // customization attributes
+      );
+    } catch (err) {
+      console.log(err);
```
```suggestion
      console.log(err);
```
is the console required?

sidarth-23 committed:
feat: Add GitHub and Google sign-in buttons

- Add GitHub sign-in button to allow users to sign in with GitHub.
- Add Google sign-in button to allow users to sign in with Google.
- These buttons will authenticate users and set cookies for access and refresh tokens.
refactor: Fix import paths for actions and api-routes (#123)
✨ Add dynamic googleId reference in GoogleSignInButton

ℹ️ Use dynamic googleId reference instead of hardcoding window.google.accounts.id in useState hooks and effects to improve code readability and maintainability.
✨ Fix useState initialization in GitHubSignInButton component
✨ Add error handling for Google Sign-in process

Added comments to handle errors during Google Sign-in process for better
error management and user experience.
✨ Add Google Sign-In button and configuration interfaces

- Added new types for Google Sign-In button and Google Sign-In function.
- Updated Google Sign-In button component to use the new interfaces.
- Updated global window declaration for Google Sign-In functionality.
✨ Add type annotations for state variables in GitHubSignInButton component
